### DISCUSSION TITLE
Apple Pay Not Show On Real IOS Device

### DISCUSSION THREAD

**Discussion by loopstack33** [flutter-stripe/flutter_stripe#1445] [Q&A]
I am unable to see apple pay on real ios device but on jos simulator its working fine. I am using flutter platform and flutter_stripe package. I have completed all steps but still apple pay does not shows on real device
Please help me
Also i am unable to process google pay it says Merchant is having trouble accepting payments right now

@remonh87

**Comment by remonh87** [comment]
I would suggest to check out https://stripe.com/docs/apple-pay. you need to create a merchant certificate for apple and upload this to apple. 

For google you cannot process the payment when Google didn't approve you: see https://stripe.com/docs/google-pay for more instructions

**Comment by loopstack33** [comment]
Brother i have done all the implementation for apple pay as mentioned. Made
a certificate uploaded to stripe, generated a merchant identifier and added
to xcode and all left but it does not works in real device shows in ios
simulator

**Comment by Kircoo** [comment]
Same issue here for iOS, I've done all the implementation/integration for Apple Pay.
It worked perfectly before, but now suddenly I am experiencing this error:

Apple pay:
LocalizedErrorMessage(code: FailureCode.Failed, localizedMessage: Payment not completed, message: Payment not completed, stripeErrorCode: null, declineCode: null, type: null)

Card payment after I add the card:
StripeException(error: LocalizedErrorMessage(code: FailureCode.Failed, localizedMessage: No payment sheet has been initialized yet, message: No payment sheet has been initialized yet, stripeErrorCode: null, declineCode: null, type: null))

This happens on real devices, the simulator works fine

**Reply by loopstack33** [reply]
@remonh87 Please help us we cannot use apple pay.. Can u create a working video with all cases covered and steps that would be helpful

**Reply by yinghuihou** [reply]
try my steps, they are completely feasible

**Comment by yinghuihou** [comment]
I encountered the same problem and finally resolved it，I encountered the correct answer here ： https://stackoverflow.com/questions/43120587/apple-pay-works-on-simulator-but-not-on-device

If you can see the apple pay option on your simulator and are confused about why it is not showing on a real device, then this answer is for you.

Initially, you may not see the apple pay option on real devices because you may not have any cards added to the Wallet app.

In this case, you have to follow [Apple Pay - Sandbox Testing - Apple Developer](https://developer.apple.com/apple-pay/sandbox-testing/) and set up a sandbox account to test apple pay.

Steps: 

1  Go to https://appstoreconnect.apple.com/access/testers and add your tester account. Use a fresh email address that was not used with any Apple services.

2  From your iPhone, go to settings, sign out from your logged-in Apple ID, and log in to your new tester account.

3  Select your region, where apple pay is available. E.j. United States (If not already)，This step is very important，This step is very important

4  Open the Wallet application and add a new card. In the first step of the document, there is a lot of test card information included

5  When successfully card added open your iOS app and you can see apple pay now.

**Reply by dicky28revo** [reply]
Hi @yinghuihou , I have many CC on the Wallet App but still not showing on real device, Do you know what's happens with it? 

It worked at iOS simulator

**Reply by yinghuihou** [reply]
Have you tried following my steps? You need to use Apple's test account to log in to your phone and add a test bank card，and don't forget to set the system time zone of your phone

**Reply by AmitBhandari7777** [reply]
Hi @yinghuihou, How is sandbox mode is related to apply pay not showing in this case??
Could you elaborate it.
I am facing the same problem.
Runs in simulator for test
But the Apple Pay does not show up in distribution

**Comment by AmitBhandari7777** [comment]
I was using Riverpod for state management, but I encountered this issue with the Stripe integration. When I moved the Stripe-related code from the RiverPod state management to the widget itself, everything started working.
